Library/Homebrew/tab.rb

```diff
@@ -45,8 +45,8 @@ def self.create(formula, compiler, stdlib)
         "tap_git_head" => nil, # Filled in later if possible
         "spec"         => formula.active_spec_sym.to_s,
         "versions"     => {
-          "stable"         => formula.stable&.version.to_s,
-          "head"           => formula.head&.version.to_s,
+          "stable"         => formula.stable&.version&.to_s,
+          "head"           => formula.head&.version&.to_s,
```

Comment:
`to_s` shouldn't be needed since `Version#to_str` exists.

Reply:
I guess `to_str` should actually error for `NULL` versions. Similar to `nil.to_s` vs `nil.to_str`.

Also `Version::NULL.to_json` should output `null` instead of `""`.

Reply:
It's going to be a string when reading from the JSON so won't exactly make it make the code simpler having to adjust for that.

But I see what you mean about `Version::NULL`, so will likely indeed need to handle it that way.

Reply:
Yeah, makes sense. I didn't check how this is used elsewhere in this file. I thought this is just a `Hash` that ends up with `to_json` being called on it.